crates/sui-benchmark/src/workloads/workload.rs

```diff
@@ -2,18 +2,66 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use async_trait::async_trait;
+use std::collections::HashMap;
 use std::sync::Arc;
 
 use crate::system_state_observer::SystemStateObserver;
 use crate::workloads::{Gas, GasCoinConfig};
 
+use crate::workloads::batch_payment::batch_payment_initializer;
+use crate::workloads::delegation::delegation_initializer;
 use crate::workloads::payload::Payload;
+use crate::workloads::shared_counter::shared_counter_initializer;
+use crate::workloads::transfer_object::transfer_object_initializer;
 use crate::ValidatorProxy;
+use once_cell::sync::Lazy;
+use strum_macros::{Display, EnumString};
 
 // This is the maximum gas we will transfer from primary coin into any gas coin
 // for running the benchmark
 pub const MAX_GAS_FOR_TESTING: u64 = 1_000_000_000;
 
+#[derive(Display, Copy, Clone, Hash, PartialEq, Eq, Debug, EnumString)]
+#[strum(serialize_all = "snake_case")]
+pub enum WorkloadType {
+    SharedCounter,
+    TransferObject,
+    Delegation,
+    BatchPayment,
+}
+
+#[derive(Display, Copy, Clone, Hash, PartialEq, Eq, Debug, EnumString)]
+#[strum(serialize_all = "snake_case")]
+pub enum WorkloadInitParameter {
```

Comment:
We should make this a enum with underlying types i.e.
```
NumTransferAccounts(u32),
SharedCounterHotnessFactor(u32),
...
```
And we can also roll in common params like max_ops, etc in this enum list as well.